spec/takeaway_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'takeaway'
+
+describe Takeaway do
+
+  subject(:takeaway) {described_class.new(menu)}
+  let(:menu) {double(:menu, :menu_list => {"chicken" => 15}) }
+
+  before do
+    allow(takeaway).to receive(:send_text)
+  end
+
+  describe '#list_menu' do
+
+    it 'lists the current available menu items' do
+      expect(takeaway.show_menu).to eq({"chicken" => 15})
+    end
+  end
+
+  describe '#order' do
+    it 'orders items from the menu' do
+      order_result = "2 * chicken added to your order"
+      expect(takeaway.order("chicken", 2)).to eq order_result
+    end
+  end
+
+  describe '#basket' do
+    it 'returns a string to the client showing their order' do
+      takeaway.order("chicken", 2)
+      test_order = "Your order summary is: \nchicken * 2 for $30,\nTotal = 30\n"
+      expect{ takeaway.basket}.to output(test_order).to_stdout
+    end
+  end
+
+  describe '#complete_order' do
+
+    it 'sends an sms to confirm the order' do
+      test_message = "Your order totalling $30 has been registered and will be delivered before 18:52"
```

Comment:
Metrics/LineLength: Line is too long. [102/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)